### ISSUE TITLE
Want annotated Interlisp bibliography

### ISSUE THREAD

**Issue by masinter** [Interlisp/medley#299]
https://scholar.archive.org/search?q=interlisp&filter_type=everything
has a lot of material that mentions Interlisp and Interlisp-D, and the History and Timeline documents have some bibliographic references.  From looking at this, there are various categories; from papers and articles ABOUT Interlisp, to citations that merely mention Interlisp in passing (the latter not so interesting.

I'd like to collect metadata and online copies (if available) of Interlisp Reference Manuals (many of them I donated to CHM several years ago.

**Comment by masinter** [comment]
@AbeJellinek wrote  https://github.com/Interlisp/Interlisp.github.io/pull/3 
I don’t think it’ll run from a file:// URL because it makes AJAX requests, but you can test it out by running:
```
$ cd /path/to/Interlisp.github.io/
$ python -m SimpleHTTPServer 8080  
```
or (python 3)
```
$ python -m http.server 8080
```



Then visit localhost:8080/bibliography/ in your browser.


Part of the way down the page, there’s one item with an annotation added so you can see what that looks like. We obviously need to clean up the collection a bit (some funny OCR artifacts in there), but it’s looking good.

**Comment by masinter** [comment]
was hoping for something where what was visible was very simple.
For example, “year” in bold, “title” as a hyperlink, “author” (if important to say), “genre”
(e.g. “book”, “article”, “publication”, with a link/icon to show (in a hover/popup).

Going to Zotero every time….. results should be cached?

**Comment by nbriggs** [comment]
How do we fix the citations?   If the current style is supposed to be IEEE, can we choose something else that gives better formatting?

**Comment by AbeJellinek** [comment]
Yes! Pretty easily. There's [an editor](https://editor.citationstyles.org/visualEditor/) to create the XML style definitions that the citation generator uses. I can create one following that spec.

**Comment by nbriggs** [comment]
I mean the source of the citations -- in Zotero?  A number of entries have garbage (e.g., bogus authors, OCR errors, ...)

**Comment by masinter** [comment]
The citations were picked up from a scan of different sources for the word “Interlisp” – scholar.archive.org (which says it is beta with poor quality metadata, but more full-text PDFs), scholar.google.com (more records), dl.acm.org (I have a lifetime membership), patents.google.com, plus, for good measure, a few of the documents we’ve collected.
I was just getting an idea of what we might do with Zotero’s current capabilities.
To get an idea of what is there, you should get Zotero desktop.


There are a bunch of people on Fiverr who offer to gather, clean up Zotero references.
The going rate seems to be about $0.50 / reference.  
https://www.fiverr.com/search/gigs?query=zotero
so maybe $100? Or sticking to sites that QA their citations ? This is all exploratory.

**Comment by masinter** [comment]
I found someone on Fiverr who has a software background to edit the Zotero bibliography.
$15 for 45 entries. Marked with the tag “sz” or “s2z”.  (sana_zafar).

Zotero doesn’t seem to have a built in facility to convert from subcollections to tags and vice versa.
I made subcollections
 “About Lisp”
“Common Lisp”
“Interlisp”
“Interlisp-D”
“Applications written in Interlisp”
“Mention Interlisp in passing”
“Irrelevant”

And

“Unsorted” – things that belong in one of the  previous categories. 

I was imagining the web site would show some but not all of these categories, sorted by Decade, within a Decade by Item Type (or some categories of item types) and dates

Anyway, please review https://www.zotero.org/groups/2914042/interlisp/library 
At some point soon we should take this as a baseline, close this issue, and open any additional work as separate issues marked "Documentation".

**Comment by masinter** [comment]
@pmcjones note

**Comment by masinter** [comment]
Status of Interlisp bibliography:
@pmcjones put together a wonderful annotated bibliography which tells a story. http://www.softwarepreservation.org/projects/LISP/interlisp_family#Interlisp-D_
and quotes my recollection with some facts I got wrong.
I had made a couple of attempts at putting together a more detailed ["History"](https://docs.google.com/document/d/1znqCtOMNYP3sb1UVv1S3iYEZIcboL9-SwTSiym6Zym8/edit#heading=h.tnawpcvoj8h6) and  ["Timeline"](https://docs.google.com/document/d/1QXUMQ0OUttYYzWUvgdVcnkz5Gym_t1MDnMT8XZJe-NI/edit#heading=h.eh7hehd7zj4s) before shifting toward a strategy of linking citations and notes, and using that to generate the https://interlisp.org/bibliography  .   I see now this is a research project. But I think it's still worth trying.

The bibliography we have [in Zotero](https://www.zotero.org/groups/2914042/interlisp/collections/FSK5IX4F) has a lot of entries somewhat at random: I did searches for "Interlisp" in https://scholar.google.com https://scholar.archive.org https://dl.acm.org. I also paid someone on Fiverr.com to "fix" the references we have by completing the metadata including abstracts, URLs and other fields. They spent more than a week on the project.

**Comment by hjellinek** [comment]
We need a precise definition of what it means to resolve (complete, close) this issue.  Given that no bibliography will ever be finished, I propose to engage a bibliographer/Zotero expert, probably on Fiverr, with carefully-worded instructions to take the current 'sz' Zotero entries, compare them with Paul McJones's annotated bibliography, and (1) flesh out the items tagged with 'sz' with the information from Paul's bibliography they do not include; (2) add the items from Paul's bibliography that are not already present in Zotero.

We'll name the result something that clearly identifies the source.

**Comment by masinter** [comment]
There were other sources for the entries that sz worked on. I spent several hours looking for "Interlisp" in scholar.google.com and scholar.archive.org. As well as patents.google.com and the links in the two Google docs linked from the History section of Interlisp.org.   It's fine to start small, but don't start with the 'sz' tagged items. Start with Paul's document; look for citations we have; tag them with a tag that indicates which sub-collection they were in, and move them to a new "checked" sub-collection.
We can publish that on our web site.

For citations from Paul's that we don't already have (I don't expect any) I'd say add them to a "New" subcolletion.

**Comment by pmcjones** [comment]
Herb,

I spent some time going over one of the subcategories of Larry’s Zotero bibliography, and filled in a few missing dates and so on. I must say Larry’s bibliography is probably much more comprehensive than mine: I was focused mostly on papers and documents by the Interlisp group about the core system, so I missed a lot of the application stuff, which is very important.

Larry,

In your message of August 23, you said:

> @pmcjones <https://github.com/pmcjones> put together a wonderful annotated bibliography which tells a story. http://www.softwarepreservation.org/projects/LISP/interlisp_family#Interlisp-D <http://www.softwarepreservation.org/projects/LISP/interlisp_family#Interlisp-D>_
> and quotes my recollection with some facts I got wrong.

If there are incorrect facts in http://www.softwarepreservation.org/projects/LISP/interlisp_family/, I’d like to correct them.


Paul

**Comment by hjellinek** [comment]
Let me rephrase what I wrote above: 

I will start (the Fiverr person) with Paul's document; look for citations we have, paying particular attention to the sz ones, which may need to be updated or enlarged; and move the result to a new collection/library.

I'm not clear on what you mean by tagging them by "what sub-collection they were in."  Also, libraries/collections can be moved under other collections/libraries at any time, so we're free to design a hierarchy we like at any time.  Or so I'm told....

**Comment by masinter** [comment]
@pmcjones re "facts I got wrong" let's just say that in the note to you that you quoted, there are things that I said with more certainty than they deserved. It was so common a failing at PARC that I can hear Jerry Elkind ask "Do you know, or are you just guessing?" (or something like that, I don't know for sure).

**Comment by masinter** [comment]
@hjellinek
> I'm not clear on what you mean by tagging them by "what sub-collection they were in

![image](https://user-images.githubusercontent.com/1116587/133199397-6f39384f-6e89-4cfe-9d2b-608a28ef6886.png)

you can see the rough categorization "Common Lisp" "See Also for" etc, what I was trying to do with sub-collections would probably be done better with tags.
